src/core/lifetime.d

```diff
@@ -2361,3 +2427,191 @@ template _d_delstructImpl(T)
     assert(innerDtors == 2);
     assert(outerDtors == 1);
 }
+
+// issue 25552
+pure nothrow @system unittest
+{
+    int i;
+    struct Nested
+    {
+    pure nothrow @nogc:
+        char[1] arr; // char.init is not 0
+        ~this() { ++i; }
+    }
+
+    Nested[1] dst = void;
+    Nested[1] src = [Nested(['a'])];
+
+    moveEmplace(src, dst);
+    assert(i == 0);
+    assert(dst[0] == Nested(['a']));
+    assert(src[0] == Nested());
+}
+
+// issue 25552
+@safe unittest
+{
+    int i;
+    struct Nested
+    {
+        void foo() { ++i; }
```

Comment:
This appears unused. If it is intentional, use it at the end and add a comment to avoid uncovered lines.

Reply:
Oh, good catch, thanks. Since my other fix for static arrays is now in this can now be moved to destructor.